internal/cloudmock/metrics.go

```diff
@@ -71,20 +82,26 @@ func (m *MetricsTestServer) CreateServiceTimeSeriesRequests() []*monitoringpb.Cr
 	return reqs
 }
 
-func (m *MetricsTestServer) appendCreateMetricDescriptorReq(req *monitoringpb.CreateMetricDescriptorRequest) {
+func (m *MetricsTestServer) appendCreateMetricDescriptorReq(ctx context.Context, req *monitoringpb.CreateMetricDescriptorRequest) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 	m.createMetricDescriptorReqs = append(m.createMetricDescriptorReqs, req)
 }
-func (m *MetricsTestServer) appendCreateTimeSeriesReq(req *monitoringpb.CreateTimeSeriesRequest) {
+func (m *MetricsTestServer) appendCreateTimeSeriesReq(ctx context.Context, req *monitoringpb.CreateTimeSeriesRequest) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 	m.createTimeSeriesReqs = append(m.createTimeSeriesReqs, req)
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		m.userAgent = strings.Join(md.Get("User-Agent"), ";")
```

Comment:
Looks like this would overwrite UA across different requests. Should be OK, maybe we should raise an error if it would be overwritten with a new value

Reply:
Do we expect there to be multiple `User-Agent` MD values or just a single one pre-joined with `;`?

Reply:
We expect a single `User-Agent` MD value. I just wanted to make sure we would notice if we somehow got more than one.

I'll update it to append to the user-agent.  That way, we would just fail, and be able to see all the user-agents that were written